### DISCUSSION TITLE
How to get dns from upstream block

### DISCUSSION THREAD

**Discussion by kowsuram** [nginx/nginx#238] [General]
I am trying to implement a load balancer based on the health check.

If one of the cluster ingress goes down it has to routed to the other cluster ingress.

I did the following

upstream backend {
 server primary max_fails=2, timeout2s;
 server secondary backup;
}

location / {
resolver dns-default.openshift-dns;
proxy_pass https://backend;
proxy_pass_request_headers on;
proxy_ssl_server_name on;

proxy_ssl_name <active_ingress_dns??>;
proxy_set_header Host <active_ingress_dns??>;
}
I stuck on how to get the active ingress dns because it is needed to set the proxy SSL name and header host

I tried with map blocks

map $upstream_addr $aidns{
 121.12.11.12 "primary.com"
 123.23.32.12 "secondary.com"
}
but $aidns is not consistent always. Sometimes it is printing the upstream name, ip with upstream name

Is there any way that I can get the domain name consistently.

**Comment by viktorberezikov** [comment]
It is assumed that all servers in an upstream group are the same. And requests proxied for balancing should be consistent. This means that you cannot set different SNIs and different HTTP headers for individual servers in an upstream group.
